src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -6141,7 +6135,7 @@ static bool M(TypedReference x, int* p, ref int z)
                 Diagnostic(ErrorCode.ERR_PatternWrongType, "object").WithArguments("System.TypedReference", "object").WithLocation(9, 23),
                 // (10,23): error CS0244: Neither 'is' nor 'as' is valid on pointer types
                 //         var b2 = p is object o2;         // not allowed 2
-                Diagnostic(ErrorCode.ERR_PointerInAsOrIs, "object o2").WithLocation(10, 23),
+                Diagnostic(ErrorCode.ERR_PointerInAsOrIs, "object").WithLocation(10, 23),
```

Comment:
>Diagnostic(ErrorCode.ERR_PointerInAsOrIs, "object").WithLocation(10, 23), [](start = 16, length = 73)

Location for this error is unexpected.

Reply:
This error is traditionally placed on the type in the right-hand-side. I'm fine with this, but will keep an eye on it for more complex patterns.

---
In reply to: [156567331](https://github.com/dotnet/roslyn/pull/23209#discussion_r156567331) [](ancestors = 156567331)